### PR TITLE
DMP-515-date-pickers

### PR DESCRIPTION
**Pull request by bpires2**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-515

### Change description ###

Addition of date pickers for date input

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/components/search/search.component.css [hunk 1]
```diff
@@ -2,3 +2,72 @@
...
+.ds_datepicker {
+  box-sizing: border-box;
+}
+
+.ds_input {
+  border: 2px solid #0b0c0c;
+  border-radius: 0;
+  display: block;
+  margin-bottom: 24px;
+  /*min-height: 40px;*/
```
Comment (davet1985) [inline src/app/components/search/search.component.css:15]: Remove this line if it's not required?

File: src/app/components/search/search.component.css [hunk 2]
```diff
@@ -2,3 +2,72 @@
   margin-left: 15px;
   padding-left: 30px;
 }
+
```
Comment (davet1985) [inline src/app/components/search/search.component.css:5]: Should all this CSS be in here? Or in a global place? What if we want to use date pickers else where?

File: src/styles.scss [hunk 3]
```diff
@@ -1,7 +1,721 @@
-/* You can add global styles to this file, and also import other style files */
 @import 'govuk-frontend/govuk/all';
 @import '@ministryofjustice/frontend/moj/all';
 
+/*------------------------------------*\
```
Comment (davet1985) [inline src/styles.scss:4]: I'd rather this was stored in a separate file and imported

File: src/assets/datepicker.scss [hunk 4]
```diff
@@ -0,0 +1,787 @@
...
+  cursor: pointer;
+  display: inline-block;
+  font-weight: 700;
+  line-height: 24px;
+  outline: 2px solid transparent;
+  outline-offset: -2px;
+  position: relative;
+  text-align: center;
+  text-decoration: none;
+  // transition: background-color .2s;
```
Comment (davet1985) [inline src/assets/datepicker.scss:106]: Remove?

File: src/assets/datepicker.scss [hunk 5]
```diff
@@ -0,0 +1,787 @@
...
+  padding: 8px;
+  width: 40px;
+}
+
+.ds_button:focus {
+  background-color: #ffdd00;
+  box-shadow: 0 3px #333;
+  color: #333;
+  outline: 0;
+  // transition-duration: 0s
```
Comment (davet1985) [inline src/assets/datepicker.scss:128]: Remove?

File: src/assets/datepicker.scss [hunk 6]
```diff
@@ -0,0 +1,787 @@
...
+      position: absolute;
+
+      &__title {
+        line-height: line-height-on-grid(16px, 1.5);
+        padding: 8px 0;
+      }
+
+      &__header {
+        > :nth-child(1) {
+          // left: 8px;
```
Comment (davet1985) [inline src/assets/datepicker.scss:688]: Remove?

File: src/assets/datepicker.scss [hunk 7]
```diff
@@ -0,0 +1,787 @@
...
+          top: 0;
+        }
+
+        > :nth-child(3) {
+          // right: 8px;
```
Comment (davet1985) [inline src/assets/datepicker.scss:693]: Remove?

**Review by DavidTMann — COMMENTED** [review]
added comment r.e. init functions

File: src/app/components/search/search.component.ts [hunk 8]
```diff
@@ -66,6 +67,10 @@ export class SearchComponent {
     this.dateInputType = type;
   }
 
+  ngAfterViewChecked(): void {
```
[the remark this answers is not in the document]
Comment (bpires2) [reply src/app/components/search/search.component.ts:70]: I tried using ngAfterViewInit and it didn't work for this problem, so ngAfterViewChecked was used

**Review by DavidTMann — APPROVED** [review]

**Pull request merged**